merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/models/ConstraintsCompilationError.java

```diff
@@ -30,7 +31,7 @@ public class ConstraintsCompilationError {
 
   public static final JsonParser<List<UserCodeError>> constraintsErrorJsonP = listP(userCodeErrorP);
 
-  public record CodeLocation(Integer line, Integer column) {}
+  public record CodeLocation(Optional<Integer> line, Optional<Integer> column) {}
```

Comment:
Do you know the case under which you found the code location to be missing? Just making sure we're not missing some root problem here

Reply:
Its during runtime errors. I added a check that the argument is greater than 0, which throws an error in the typescript.

Reply:
Based on the test case here, https://github.com/NASA-AMMOS/aerie-ts-user-code-runner/blob/develop/test/UserCodeRunner.spec.ts#L11-L45 I would expect that the code location should be returned. Can you help me recreate the case where you saw this so that I can ensure that we are indeed returning a code location?

Reply:
And as a principle, I think that ANY Error coming from the code runner through normal operation should return a location. The exceptions are when we incorrectly configure the supporting files (like what happened recently with the mission model generated code for the scheduler), there is a legitimate bug in the code runner, or there is some truly exceptional nodejs runtime failure

Reply:
If you revert the changes to `ConstraintsCompilationError.java` to remove the `Optional` ints and `nullable` parsers, the `testSplitArgumentError` test will fail. This is the test to make sure that it correctly throws an error if the argument is less than 1.

The test fails because it cannot parse the error, because its coming in with `{line: null, column: null}`. I'm not sure why it doesn't have location, but this is the code that produces the error:

```ts
public split(numberOfSubWindows: number): Windows {
    if (numberOfSubWindows < 1) {
      throw RangeError(".split numberOfSubWindows cannot be less than 1, but was: " + numberOfSubWindows);
    }
    return new Windows({
      kind: AST.NodeKind.IntervalsExpressionSplit,
      intervals: this.__astNode,
      numberOfSubIntervals: numberOfSubWindows
    })
  }
```

Reply:
Ah, I don't have a regression test for errors thrown in additional files. I'll add one and ensure we properly map back the errors to the location in the users code

Reply:
Ok, I'll remove the `Optional`'s then. I'm still going to keep the `nullable` parser implementation though, for when nulls come up again. (I think we'll end up using it for external profiles.)

Reply:
https://jira.jpl.nasa.gov/browse/AERIE-1987 for reference.

Easy to debug, easy to fix. Should have it PR'd for you within the hour.